object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/BooleanFilterNormalizer.java

```diff
@@ -176,6 +192,134 @@ private int comparePrimaryPredicateExpr(boolean isFirstNegated, PrimaryPredicate
          return -1;
       }
 
+      private void optimizeOverlappingIntervalPredicates(List<BooleanExpr> children, boolean isConjunction) {
+         for (int i = 0; i < children.size(); i++) {
+            BooleanExpr ci = children.get(i);
+            if (ci instanceof ComparisonExpr) {
+               ComparisonExpr first = (ComparisonExpr) ci;
+               assert first.getLeftChild() instanceof PropertyValueExpr;
+               assert first.getRightChild() instanceof ConstantValueExpr;
```

Comment:
Hmmm, so you don't support a condition like `field = x or 1 = 1`? I remember using this trick a lot with SQL parametrized statements, people might use it for query as well.